arkouda/alignment.py

```diff
@@ -172,6 +175,69 @@ def in1dmulti(a, b, assume_unique=False, symmetric=False):
             return atruth
 
 
+def find(query, space):
+    """
+    Return indices of query items in a search list of items (-1 if not found).
+
+    Parameters
+    ----------
+    query : (sequence of) array-like
+        The items to search for. If multiple arrays, each "row" is an item.
+    space : (sequence of) array-like
+        The set of items in which to search. Must have same shape/dtype as query.
+
+    Returns
+    -------
+    indices : pdarray, int64
+        For each item in query, its index in space or -1 if not found.
+    """
+
+    # Concatenate the space and query in fast (block interleaved) mode
+    if isinstance(query, (pdarray, Strings, Categorical)):
+        if type(query) != type(space):
+            raise TypeError("Arguments must have same type")
+        c = concatenate((space, query), ordered=False)
+        spacesize = space.size
+        querysize = query.size
+    else:
+        if len(query) != len(space):
+            raise TypeError("Multi-array arguments must have same number of arrays")
+        spacesize = set(s.size for s in space)
+        querysize = set(q.size for q in query)
+        if len(spacesize) != 1 or len(querysize) != 1:
+            raise TypeError("Multi-array arguments must be non-empty and have equal-length arrays")
+        spacesize = spacesize.pop()
+        querysize = querysize.pop()
+        atypes = np.array([ai.dtype for ai in query])
+        btypes = np.array([bi.dtype for bi in space])
```

Comment:
Noting that this will break if anything is Categorical because `.dtype` is not yet supported.